### PR TITLE
Porting Microsoft.Build to .NET Core

### PR DESCRIPTION
**Pull request by dsplaisted**
Progress towards getting Microsoft.Build to compile for .NET Core

### CODE REVIEW

File: src/XMakeBuildEngine/BackEnd/Components/RequestBuilder/RequestBuilder.cs [hunk 1]
```diff
@@ -640,19 +640,23 @@ private void SetCommonWorkerThreadParameters()
...
             Thread.CurrentThread.IsBackground = true;
 
+            // NOTE: This is safe to do because we have specified long-running so we get our own new thread.
+            string threadName = "RequestBuilder thread";
+
+#if FEATURE_APARTMENT_STATE
             if (Thread.CurrentThread.GetApartmentState() == ApartmentState.STA)
             {
                 // NOTE: This is safe to do because the STA scheduler always gives us our own new thread.
                 Thread.CurrentThread.Name = "RequestBuilder STA thread";
```
Comment (jonorossi) [inline src/XMakeBuildEngine/BackEnd/Components/RequestBuilder/RequestBuilder.cs]: Shouldn't this be `threadName = "RequestBuilder STA thread";`?

Comment (dsplaisted) [inline src/XMakeBuildEngine/BackEnd/Components/RequestBuilder/RequestBuilder.cs]: > Shouldn't this be `threadName = "RequestBuilder STA thread";`?
Yes, good catch!

File: src/Shared/AssemblyNameExtension.cs [hunk 2]
```diff
@@ -722,12 +737,16 @@ internal bool CompareCulture(AssemblyNameExtension that)
...
             }
+
             if (aCulture.LCID != bCulture.LCID)
             {
                 return false;
             }
 
             return true;
+#else
+            return CultureInfo?.Name == that.CultureInfo?.Name;
```
Comment (AndyGerlicher) [inline src/Shared/AssemblyNameExtension.cs]: Guess we're going C# 6 now :)

**Comment by AndyGerlicher** [comment]
LGTM

File: src/Shared/AssemblyNameExtension.cs (hunk 2 above)
Comment (dsplaisted) [inline src/Shared/AssemblyNameExtension.cs]: Unless anyone has any objections... :-)

**Pull request merged**